Index: Homework.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>#_____________________ Задание 1 ()\r\nprint('Задание №1 ↓')\r\n\r\n#_____________________ Задание 2 ()\r\nprint('Задание №2 ↓')\r\n\r\n#_____________________ Задание 3 ()\r\nprint('Задание №3 ↓')\r\n\r\n#_____________________ Задание 4 ()\r\nprint('Задание №4 ↓')\r\n\r\n#_____________________ Задание 5 ()\r\nprint('Задание №5 ↓')\r\n\r\n#_____________________ Задание 6 ()\r\nprint('Задание №6 ↓')
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Homework.py b/Homework.py
--- a/Homework.py	
+++ b/Homework.py	
@@ -1,17 +1,142 @@
-#_____________________ Задание 1 ()
+# _____________________ Задание 1 (Создать функцию с двумя аргументами и выполнить деление)
 print('Задание №1 ↓')
 
-#_____________________ Задание 2 ()
+
+def input_values():
+    """ Принимает значения двух чисел, выполняет проверку 'Знаменатель == 0'"""
+
+    num_1 = int(input('Введите числитель: '))
+    num_2 = int(input('Введите знаменатель: '))
+    if num_2 == 0:
+        print('На ноль делить нельзя, давайте попробуем ещё раз')
+        return input_values()
+    else:
+        return div(num_1, num_2)
+
+
+def div(num_1, num_2):
+    """ Возвращает результат деления двух чисел
+
+    (number, number) -> number
+
+    >>> div(4, 2)
+    2
+    """
+    return num_1 / num_2
+
+
+print(f'Результат деления: {round(input_values(), 2)}', end='\n' * 2)
+
+# _____________________ Задание 2 (Ввести и вывести данные пользователя через функцию)
 print('Задание №2 ↓')
 
-#_____________________ Задание 3 ()
+
+def user_items(name, surname, birth, city, email, phone):
+    """ Принимает и выводит информацию о пользователе """
+
+    print(f'Пользователь {surname.capitalize()} {name.capitalize()}, '
+          f'родился {birth}, проживает в городе {city.capitalize()}, '
+          f'e-mail : {email}, номер телефона: {phone}', end='\n' * 2)
+
+
+n = input('Введите имя пользователя : ')
+s = input('Введите фамилию : ')
+b = input('Введите дату рождения : ')
+c = input('Введите город проживания : ')
+e = input('Введите email : ')
+p = input('Введите номер телефона : ')
+user_items(name=n, surname=s, phone=p, email=e, city=c, birth=b)
+
+
+
+# _____________________ Задание 3 (Функция с тремя аргументами и суммой двух наибольших значений)
 print('Задание №3 ↓')
 
-#_____________________ Задание 4 ()
+
+def my_func(num1, num2, num3):
+    """ Возвращает сумму двух наибольших чисел
+
+    (number, number, number) -> number
+
+    >>> my_func(4, 2)
+    2
+    """
+    numbers = sorted([num1, num2, num3], reverse=True)
+    return f'Сумма двух наибольших значений = {sum(numbers[:2])}'
+
+
+print(my_func(float(input('Введите первое число : ')),
+              float(input('Введите второе число : ')),
+              float(input('Введите третье число : '))
+              ), end='\n' * 2)
+
+# _____________________ Задание 4 (Принять два аргумента и возвести х в степень у двумя методами без ф-ии pow)
 print('Задание №4 ↓')
 
-#_____________________ Задание 5 ()
+def my_func(x, y):
+    """ Возвращает число, возведённое в степень
+
+    (number, number) -> number
+
+    >>> my_func(4, 2)
+    16
+    """
+    numxy = x
+    for i in range(abs(y) - 1):
+        numxy *= x
+    if y < 0:
+        numxy = 1 / numxy
+    elif y == 0:
+        numxy = 1.0
+
+    return f" (Решение 1) Число {x} в степени {y} = {round(x**y, 3)} \n" \
+           f" (Решение 2) Число {x} в степени {y} = {round(numxy, 3)}"
+
+
+numx, numy = float(input('Введите число : ')), int(input('Введите степень : '))
+print(my_func(numx, numy), end='\n' * 2)
+
+
+# _____________________ Задание 5 (Выводить сумму чисел, пока пользователь не введёт спец.символ)
 print('Задание №5 ↓')
 
-#_____________________ Задание 6 ()
-print('Задание №6 ↓')
\ No newline at end of file
+print(' Введите строку чисел, разделённых пробелом. Программа выведет сумму этих чисел. Если вы хотите '
+      'закончить ввод, введите символ "%"')
+number_sum = 0
+flag = True
+
+while flag:
+    ls = input('Введите строку : ').split()
+    if '%' in ls:
+        flag = False
+        ls.remove('%')
+    ls = [int(i) for i in ls]
+    number_sum += sum(ls)
+    print(f'Сумма всех введённых элементов = {number_sum}')
+print(f'Работа программы завершена', end='\n' * 2)
+
+
+#_____________________ Задание 6 ( capitalize слово)
+print('Задание №6 ↓')
+
+
+def int_func(txt):
+    """ Делает первую букву слова заглавной"""
+    return txt.capitalize()
+
+
+print(f" Ваше слово с заглавной буквы : {int_func(input('Введите слово: '))}", end="\n" * 2)
+
+#_____________________ Задание 7 (Делаем каждую первую букву слов строки заглавной)
+print('Задание №7 ↓')
+
+
+def int_func(txt):
+    """ Делает первую букву слова заглавной"""
+    return txt.capitalize()
+
+
+sentence = input('Введите предложение : ').split()
+for i in range(len(sentence)):
+    sentence[i] = int_func(sentence[i])
+print(*sentence)
